### ISSUE TITLE
A11y: Combobox does not behave correctly on screenreader

### ISSUE THREAD

**Issue by idfunctor** [ariakit/ariakit#2708]
### Current behavior

As we can see in the screenrecording, Combobox items are always read as "selected" by Apple Voiceover. This is not the case with Select component, where all items are read as "not selected" except for the selected one.

One thing to note, is that "Select with combobox" example also has this bug (refer screenshot)

All in all, it is fair to say that this bug surfaces whenever we use ComboboxItem, with or without Select.

https://github.com/ariakit/ariakit/assets/13308108/2af275e8-56bf-47f7-a293-90c59891b331

<img width="926" alt="image" src="https://github.com/ariakit/ariakit/assets/13308108/0be089a3-5786-4331-a7ed-a57c64aa9340">

### Steps to reproduce the bug

1. Open any combobox example on ariakit docs
2. Turn on voiceover
4. try to navigate between items
5. compare the behaviour with WCAG combobox examples, MUI Autocomplete, etc or with Select of Ariakit itself.

You will notice that in ariakit combobox, all items are read as "Selected".

### Expected behavior

Only the selected items should be announced as "selected"

### Workaround

https://github.com/ariakit/ariakit/issues/2708#issuecomment-1692000653

### Possible solutions

I am happy to work on the solution as it is important for my project, but I have no idea what is wrong because data-active-item and aria-selected properties are being applied correctly, and are same between Select and Combobox. If I am helped with identifying the bug I would be happy to help fix it.

Moreover, I would like to explore if it's a good idea to add https://www.guidepup.dev/ or some other tool to add screenreader testing to the components and avoid regressions

**Comment by idfunctor** [comment]
@diegohaz thanks a lot! should I submit a PR to change combobox with select code example to reflect the expected behaviour? Given that the “trigger” there is Select and the combobox in inside the dropdown, I would assume the screenreader behaviour has to match Select UX.

**Comment by idfunctor** [comment]
I mean this example https://ariakit.org/examples/select-combobox

**Comment by diegohaz** [comment]
> @diegohaz thanks a lot! should I submit a PR to change combobox with select code example to reflect the expected behaviour? Given that the “trigger” there is Select and the combobox in inside the dropdown, I would assume the screenreader behaviour has to match Select UX.

Yes, I think that makes sense. Feel free to submit a PR. Thank you!

**Comment by diegohaz** [comment]
> I mean this example https://ariakit.org/examples/select-combobox

Closing as the example has been fixed.
